tests/vm/buttonservo.spec.ts

```diff
@@ -37,45 +39,78 @@ suite("button servo", () => {
             const runner = new VMProgramRunner(roleMgr, program)
             await runner.startAsync()
 
-            await testBody(tester, button, servo)
+            const servoReg = new RegisterTester(
+                servo.service.register(ServoReg.Angle)
+            )
+            await testBody(tester, button.server, servo.server, servoReg)
         })
     }
 
     test(
         "inverts when pressed",
-        makeVmTest(async (tester, button, servo) => {
-            servo.server.angle.setValues([50])
+        makeVmTest(async (tester, button, servo, servoReg) => {
+            servo.angle.setValues([50])
 
-            button.server.down()
-            await tester.waitForDelay(100)
-            button.server.up()
-            assert(servo.server.angle.values()[0] == -50.0)
+            await tester.assertWith(
+                servoReg
+                    .onUpdate({
+                        triggerRange: [-50, -50],
+                    })
+                    .hold(),
+                async () => {
+                    button.down()
+                    await tester.waitForDelay(100)
+                    button.up()
+                    await tester.waitForDelay(100)
+                }
+            )
         })
     )
 
     test(
         "does not invert when not pressed",
-        makeVmTest(async (tester, button, servo) => {
-            servo.server.angle.setValues([50])
+        makeVmTest(async (tester, button, servo, servoReg) => {
+            servo.angle.setValues([50])
 
-            await tester.waitForDelay(100)
-            assert(servo.server.angle.values()[0] == 50)
+            await tester.assertWith(servoReg.hold([50, 50]), async () => {
```

Comment:
what is the meaning of ``[50,50]``?

Reply:
[50, 50] is meant as a range, but I think it will probably be a better API to allow it to take either a single number (exact match) in addition to the interval it currently supports.